usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -747,12 +747,24 @@ def add_data_dictionary_to_zip(working_dir, zip_file_path):
         try:
             RetrieveFileFromUri(data_dictionary_url).copy(data_dictionary_file_path)
             break
-        except (HTTPError, RemoteDisconnected):
+        except (HTTPError, RemoteDisconnected) as e:
+            logger.info(
+                f"Attempt {attempt + 1} of {retry_count + 2} failed to download Data Dictionary. Error: {e}"
+            )
             if attempt < retry_count:
                 time.sleep(settings.DATA_DICTIONARY_DOWNLOAD_RETRY_COOLDOWN)
                 continue
             else:
+                logger.error(f"Failed to download Data Dictionary after {retry_count + 1} attempts.")
                 raise
+        except Exception as e:
```

Comment:
What's the reason you do not want to retry in case of exceptions other than the types `HTTPError, RemoteDisconnected`?

Reply:
It's a good practice to plan for specific types of Exceptions and how to handle them. In this case, we're expecting the `RetrieveFileFromUri()` function to throw some sort of network error. That's really the only reason why we'd want to try another request.

I added the extra catch (`except`) because some sort of error seems to be slipping through. In this case of a unexpected error, we should log and raise it, so it can be addressed. 

Here are a few examples of exceptions we wouldn't want to retry:
1. `ValueError`: This might be raised if the provided URI is malformed or if some other parameter value is out of an expected range. Retrying won't fix a malformed URI.
3. `PermissionError`: This could be raised if the function tries to save a file to a local directory without appropriate permissions.
4. `MemoryError`: Raised when an operation runs out of memory. If the file is too large to fit into memory, for instance, retrying is unlikely to help unless there are external factors affecting available memory.

Reply:
Ok makes sense.